independent-projects/qute/core/src/main/java/io/quarkus/qute/UserTagSectionHelper.java

```diff
@@ -28,8 +30,27 @@ public UserTagSectionHelper(Supplier<Template> templateSupplier, Map<String, Exp
             if (t1 != null) {
                 result.completeExceptionally(t1);
             } else {
-                // Execute the template with the params as the root context object
+
                 try {
+
+                    // Execute nested code for later usage in the user tag
+                    CompletableFuture<ResultNode> nestedResult = new CompletableFuture<>();
+                    context.execute(context.resolutionContext().createChild(r1, null)).whenComplete((r2, t2) -> {
+                        if (t2 != null) {
+                            result.completeExceptionally(t2);
+                        } else {
+                            nestedResult.complete(r2);
+                        }
+                    });
+                    if (result.isCompletedExceptionally()) {
+                        return;
+                    }
+
+                    StringBuilder sb = new StringBuilder();
+                    nestedResult.get().process((s) -> sb.append(s));
```

Comment:
Well, this is not a correct approach as the invocation of `CompletableFuture.get()` blocks. It should not be a big problem in most cases but we do attempt to keep the "resolution" part of rendering non-blocking. I need to think about this a little. It seems that our API does not cover this specific use case...

Reply:
Yes, I was pretty sure of that... (check my first comment) But I didn't not find the right way to do it. Note that the PR's only goal is a tentative to add a feature ; if it cannot be done correctly we should not lose too much time on it.

Reply:
The API is still evolving... so I think that we should spend some time to try to implement this (and improve the API ;-)